lib/new_relic/agent/log_event_aggregator.rb

```diff
@@ -0,0 +1,209 @@
+# encoding: utf-8
+# This file is distributed under New Relic's license terms.
+# See https://github.com/newrelic/newrelic-ruby-agent/blob/main/LICENSE for complete details.
+
+require 'new_relic/agent/event_aggregator'
+require 'new_relic/agent/log_priority'
+
+module NewRelic
+  module Agent
+    class LogEventAggregator < EventAggregator
+
+      # Per-message keys
+      LEVEL_KEY = "level".freeze
+      MESSAGE_KEY = "message".freeze
+      TIMESTAMP_KEY = "timestamp".freeze
+      PRIORITY_KEY = "priority".freeze
+
+      # Metric keys
+      LINES = "Logging/lines".freeze
+      DROPPED_METRIC = "Logging/Forwarding/Dropped".freeze
+      SEEN_METRIC = "Supportability/Logging/Forwarding/Seen".freeze
+      SENT_METRIC = "Supportability/Logging/Forwarding/Sent".freeze
+      METRICS_SUPPORTABILITY_FORMAT = "Supportability/Logging/Metrics/Ruby/%s".freeze
+      FORWARDING_SUPPORTABILITY_FORMAT = "Supportability/Logging/Forwarding/Ruby/%s".freeze
+      DECORATING_SUPPORTABILITY_FORMAT = "Supportability/Logging/LocalDecorating/Ruby/%s".freeze
+
+      named :LogEventAggregator
+      buffer_class PrioritySampledBuffer
+
+      # TODO use the right value when the collector starts reporting it
+      capacity_key :'custom_insights_events.max_samples_stored'
+      #capacity_key :'log_sending.max_samples_stored'
+      enabled_key :'application_logging.forwarding.enabled'
+
+      # Config keys
+      OVERALL_ENABLED_KEY = :'application_logging.enabled'
+      METRICS_ENABLED_KEY = :'application_logging.metrics.enabled'
+      FORWARDING_ENABLED_KEY = enabled_keys.first
```

Comment:
I hadn't added the overall `application_logging.enabled` key into actual usage (despite having added the constant) so that's part of the weirdness here.

When we wire that up, we can consider whether that's the right thing to use for `enabled_key`. I currently use `enabled?` in the middle of `record` below _after_ metrics but _before_ capturing the log, so that would need to change. Worth evaluating what would make this all the most consistent.

Also of note, my setting `enabled_key :'application_logging.forwarding.enabled'` and then later using `enabled_keys.first` to get at it is dancing around the orphaned config checking, as it recognizes the first form but doesn't follow constants. (Can chat if this doesn't make sense, it's a little weird and subtle).